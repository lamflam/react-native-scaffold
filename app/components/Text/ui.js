import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Text as RNText } from 'react-native';
import { styled } from './styles';

export class Text extends PureComponent {
    static propTypes = {
        styles: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.object])).isRequired,
        style: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
        bold: PropTypes.bool
    };

    static defaultProps = {
        bold: false,
        style: null
    };

    render() {
        const { bold, children, styles, style, ...rest } = this.props;
        return (
            <RNText style={[bold ? styles.bold : styles.text, style]} {...rest}>
                {children}
            </RNText>
        );
    }
}

export const TextUI = styled(Text);
