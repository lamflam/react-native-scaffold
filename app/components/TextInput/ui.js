import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { TextInput as RNTextInput } from 'react-native';
import { styled } from './styles';

class TextInput extends PureComponent {
    static propTypes = {
        styles: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.object])).isRequired,
        style: PropTypes.oneOfType([PropTypes.array, PropTypes.number, PropTypes.object])
    };

    static defaultProps = {
        style: null
    };

    render() {
        const { styles, style, ...rest } = this.props;
        return <RNTextInput style={[styles.input, style]} {...rest} />;
    }
}

export const TextInputUI = styled(TextInput);
