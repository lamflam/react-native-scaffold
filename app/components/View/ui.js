import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { View as RNView } from 'react-native';
import { styled } from './styles';

export class View extends PureComponent {
    static propTypes = {
        styles: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.object])).isRequired,
        style: PropTypes.oneOfType([PropTypes.number, PropTypes.object])
    };

    static defaultProps = {
        style: null
    };

    render() {
        const { children, styles, style, ...rest } = this.props;
        return (
            <RNView style={[styles.view, style]} {...rest}>
                {children}
            </RNView>
        );
    }
}

export const ViewUI = styled(View);
