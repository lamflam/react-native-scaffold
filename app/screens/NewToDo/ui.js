import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { View, Text, TextInput } from 'app/components';
import { styled } from './styles';

class NewToDo extends PureComponent {
    static propTypes = {
        styles: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.object])).isRequired,
        updateTodo: PropTypes.func,
        todo: PropTypes.string,
        addToDo: PropTypes.func
    };

    static defaultProps = {
        todo: '',
        updateTodo: () => {},
        addToDo: () => {}
    };

    render() {
        const { addToDo, updateTodo, styles, todo } = this.props;
        return (
            <View style={styles.container}>
                <View style={styles.innerContainer}>
                    <TextInput placeholder="Add todo" style={styles.todoInput} onChangeText={updateTodo} value={todo} />
                </View>
                <Text style={styles.button} onPress={addToDo}>
                    Save
                </Text>
            </View>
        );
    }
}

export const NewToDoUI = styled(NewToDo);
