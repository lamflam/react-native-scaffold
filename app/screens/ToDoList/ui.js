import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'app/components';
import { styled } from './styles';

export class ToDoList extends PureComponent {
    static propTypes = {
        newToDo: PropTypes.func,
        styles: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.object])).isRequired,
        todos: PropTypes.arrayOf(
            PropTypes.shape({
                text: PropTypes.string,
                done: PropTypes.bool
            })
        )
    };

    static defaultProps = {
        newToDo: () => {},
        todos: []
    };

    render() {
        const { newToDo, styles, todos } = this.props;
        return (
            <View style={styles.container}>
                <View style={styles.listContainer}>
                    {todos.map(todo => (
                        <Text key={todo.text} style={[styles.todo, todo.done && styles.done]}>
                            {todo.text}
                        </Text>
                    ))}
                </View>
                <Text style={styles.button} onPress={newToDo}>
                    Add todo
                </Text>
            </View>
        );
    }
}

export const ToDoListUI = styled(ToDoList);
