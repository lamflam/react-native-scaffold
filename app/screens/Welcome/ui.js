import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';
import { styled } from './styles';

export class Welcome extends PureComponent {
    static propTypes = {
        styles: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.object])).isRequired,
        toNextScreen: PropTypes.func.isRequired
    };

    render() {
        const { styles, toNextScreen } = this.props;
        return (
            <View style={styles.container}>
                <Text style={styles.welcome}>Welcome to React Native Scaffolding!</Text>
                <Text onPress={toNextScreen} style={styles.instructions}>
                    Go to my To Do List
                </Text>
            </View>
        );
    }
}

export const WelcomeUI = styled(Welcome);
