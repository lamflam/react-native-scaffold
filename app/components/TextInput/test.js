import React from 'react';
import { testSnapshot } from 'app/utils/tests';
import { TextInputUI } from './ui';

describe('TextInput UI', () => {
    it('renders correctly', () => {
        testSnapshot(<TextInputUI />);
    });
});
