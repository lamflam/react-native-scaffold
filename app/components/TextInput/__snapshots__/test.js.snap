// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput UI renders correctly 1`] = `
<TextInput
  css={[Function]}
  style={null}
  styles={
    Object {
      "input": Object {
        "backgroundColor": "transparent",
        "fontFamily": "Avenir Book",
        "fontSize": 17.5,
      },
    }
  }
  theme={
    Object {
      "colors": Object {
        "black": "#000",
        "gray00": "#111",
        "gray02": "#222",
        "gray03": "#333",
        "white": "#FFF",
      },
      "device": Object {
        "height": 1334,
        "width": 750,
      },
      "fonts": Object {
        "bold": Object {
          "backgroundColor": "transparent",
          "fontFamily": "Avenir Book",
          "fontSize": 17.5,
          "fontWeight": "900",
        },
        "normal": Object {
          "backgroundColor": "transparent",
          "fontFamily": "Avenir Book",
          "fontSize": 17.5,
        },
      },
      "normalize": [Function],
    }
  }
/>
`;
